puppetboard/app.py

```diff
@@ -94,11 +116,20 @@ def server_error(e):
     return render_template('500.html'), 500
 
 
-@app.route('/')
-def index():
+@app.route('/', defaults={'env': 'production'})
+@app.route('/<env>/')
+def index(env):
     """This view generates the index page and displays a set of metrics and
     latest reports on nodes fetched from PuppetDB.
+
+    :param env: Search for nodes in this (Catalog and Fact) environment
+    :type env: :obj:`string`
     """
+    envs = environments()
+
+    if env not in envs:
+        return redirect(url_for_environments(envs[0]))
```

Comment:
This should give a 404 imho. You're requesting an environment that's not there. Just blindly redirecting to the first entry in `envs` is surprising.